### PR TITLE
Use calendar_events api for syllabus summary

### PR DESCRIPTION
**Pull request by armstrongnate**
Fixed a bug where not all of the assignments would load because
pagination is broken for observers on the assignments api.

refs: MBL-13967
affects: parent
release note: none

Test plan:
* View a course summary and make sure all the assignments and events load
* twilson > observer1 > Assignment Overrides > Summary

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021815&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021816&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/sncd9na) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fsncd9na)
teacher: [Link](https://rebrand.ly/942gn5t) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F942gn5t)
parent: [Link](https://rebrand.ly/s71sdyw) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fs71sdyw)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/Syllabus/SyllabusSummaryViewControllerTests.swift
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-13967](https://instructure.atlassian.net/browse/MBL-13967)

<details><summary>:x: JS tests</summary>


    Summary of all failing tests
     FAIL  src/modules/submissions/list/__tests__/get-submissions-props.test.js
      ● GetSubmissionsProps gradeProp › statusProp › considers worflow unsubmitted missing
    
        expect(received).toEqual(expected) // deep equality
    
        Expected: "missing"
        Received: "none"
    
          227 | 
          228 |       const dueDate = new Date().toUTCString()
        > 229 |       expect(statusProp(submission, dueDate)).toEqual('missing')
              |                                               ^
          230 |     })
          231 | 
          232 |     it('considers pending_review submitted', () => {
    
          at Object.toEqual (src/modules/submissions/list/__tests__/get-submissions-props.test.js:229:47)
    
    
    Test Suites: 1 failed, 279 passed, 280 total
    Tests:       1 failed, 2436 passed, 2437 total
    Snapshots:   564 passed, 564 total
    Time:        274.821s
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021981&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021982&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/Syllabus/SyllabusSummaryViewController.swift [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        cell.backgroundColor = .named(.backgroundLightest)
+        cell.itemNameLabel?.text = item?.title
+        cell.iconImageView?.image = item?.type == .assignment ? .icon(.assignment, .line) : .icon(.calendarMonth, .line)
+        cell.iconImageView?.tintColor = course.first?.color
+        cell.dateLabel?.text = item?.startAt.flatMap(formatDate(_:)) ?? NSLocalizedString("No Due Date", bundle: .core, comment: "")
+        return cell
+    }
+
+    override public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let item = summary[indexPath.row]
```
Comment (bluwave) [inline Core/Core/Syllabus/SyllabusSummaryViewController.swift]: No need for row here.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Core/Core/Syllabus/SyllabusSummaryViewController.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    override public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeue(SyllabusSummaryItemCell.self, for: indexPath)
+        let item = summary[indexPath.row]
+        cell.backgroundColor = .named(.backgroundLightest)
+        cell.itemNameLabel?.text = item?.title
+        cell.iconImageView?.image = item?.type == .assignment ? .icon(.assignment, .line) : .icon(.calendarMonth, .line)
+        cell.iconImageView?.tintColor = course.first?.color
+        cell.dateLabel?.text = item?.startAt.flatMap(formatDate(_:)) ?? NSLocalizedString("No Due Date", bundle: .core, comment: "")
```
[the remark this answers is not in the document]
Comment (armstrongnate) [reply Core/Core/Syllabus/SyllabusSummaryViewController.swift]: Nope, `flatMap` is still a thing. `compactMap` is used on a sequence. This is the use of `flatMap` on an optional. This does a good job at explaining the difference https://www.avanderlee.com/swift/compactmap-flatmap-differences-explained/

File: Core/Core/Syllabus/SyllabusSummaryViewController.swift (hunk 1 above)
Comment (armstrongnate) [reply Core/Core/Syllabus/SyllabusSummaryViewController.swift]: Thanks. Fixed.

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1022115&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1022118&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1022154&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1022283&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**